### PR TITLE
kafka: versions cleanup and drop older

### PR DESCRIPTION
**Pull request by srhb**
## Description of changes

Adds new version, cleans up everything not necessary to have a sane upgrade experience on NixOS.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by srhb** [comment]
For clarity, my intention is to ignore the pkgs/by-name check failure. :smile_cat:

**Comment by infinisil** [comment]
Please don't! Check out https://github.com/NixOS/nixpkgs/pull/285922#issuecomment-1925940231 for my recommendation on what to do

**Comment by srhb** [comment]
> Please don't! Check out [#285922 (comment)](https://github.com/NixOS/nixpkgs/pull/285922#issuecomment-1925940231) for my recommendation on what to do

No problem and thanks for the advice. How does it look to you now? :)

**Comment by infinisil** [comment]
Looking good regarding `pkgs/by-name`! Make sure to create aliases for the removed attributes in `aliases.nix` though, otherwise people won't know why they're removed.

**Comment by jpds** [comment]
@srhb Hey, could you please rebase this against master and then cherry-pick my three commits from https://github.com/jpds/nixpkgs/commits/kafka-versions-cleanup-3.7/ ?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -18328,14 +18328,12 @@ with pkgs;
...
-  apacheKafka_3_0 = callPackage ../servers/apache-kafka { majorVersion = "3.0"; };
-  apacheKafka_3_1 = callPackage ../servers/apache-kafka { majorVersion = "3.1"; };
-  apacheKafka_3_2 = callPackage ../servers/apache-kafka { majorVersion = "3.2"; };
-  apacheKafka_3_3 = callPackage ../servers/apache-kafka { majorVersion = "3.3"; };
-  apacheKafka_3_4 = callPackage ../servers/apache-kafka { majorVersion = "3.4"; };
-  apacheKafka_3_5 = callPackage ../servers/apache-kafka { majorVersion = "3.5"; };
+  apacheKafkaVersions = callPackages ../servers/apache-kafka { };
+  inherit (apacheKafkaVersions)
+    apacheKafka_3_5
+    apacheKafka_3_6;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:18334]:
```suggestion
  inherit (callPackages ../servers/apache-kafka { })
    apacheKafka_3_5
    apacheKafka_3_6;
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by srhb** [comment]
@jpds Done, how does this look? :smile_cat:

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (srhb) [reply pkgs/top-level/all-packages.nix]: Done!

**Comment by jpds** [comment]
Looks great to me (though I do not have commit privileges to merge this myself).

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**